### PR TITLE
feat(make): templates and target vars

### PR DESCRIPTION
**Pull request by luphaz**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- This PR aims to use make capabilities to reduce duplication:
  - [functions](https://www.gnu.org/software/make/manual/html_node/File-Name-Functions.html)
  - [substitutions](https://www.gnu.org/savannah-checkouts/gnu/make/manual/html_node/Text-Functions.html)
  - [target specific variables](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html)
  - And templating 🙈 

NB: this was mostly to experiment what we can do with make, it might not be more maintainable this way, let me know WDYT? 

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,23 +37,65 @@ else
...
+_handler:;
+_manager: generate
+
+_docker-build-injector: docker-build-ebpf
+_docker-build-handler:;
+_docker-build-manager:;
+
+# we define the template we expect for each target
+define TARGET_template
+$(1): BINARY_NAME=$(1)
```
Comment (luphaz) [inline Makefile:65]: `$(1)` is the target name `injector|handler|manager`

File: Makefile [hunk 2]
```diff
@@ -37,23 +37,65 @@ else
...
+docker-build-ebpf:
+	docker build --platform linux/$(OS_ARCH) --build-arg ARCH=$(OS_ARCH) -t ebpf-builder-$(OS_ARCH) -f bin/ebpf-builder/Dockerfile ./bin/ebpf-builder/
+	-rm -r bin/injector/ebpf/
+	docker run --rm -v $(shell pwd):/app ebpf-builder-$(OS_ARCH)
+
+lima-push-injector lima-push-handler lima-push-manager: FAKE_FOR=COMPLETION
+
+_injector:;
+_handler:;
+_manager: generate
```
Comment (luphaz) [inline Makefile:57]: As we use a template, the template must be valid for all targets, we hence have empty targets when no dependencies before build

File: Makefile [hunk 3]
```diff
@@ -37,23 +37,65 @@ else
...
+docker-build-injector: IMAGE_TAG=$(INJECTOR_IMAGE)
+docker-build-handler: IMAGE_TAG=$(HANDLER_IMAGE)
+docker-build-manager: IMAGE_TAG=$(MANAGER_IMAGE)
+
+docker-build-ebpf:
+	docker build --platform linux/$(OS_ARCH) --build-arg ARCH=$(OS_ARCH) -t ebpf-builder-$(OS_ARCH) -f bin/ebpf-builder/Dockerfile ./bin/ebpf-builder/
+	-rm -r bin/injector/ebpf/
+	docker run --rm -v $(shell pwd):/app ebpf-builder-$(OS_ARCH)
+
+lima-push-injector lima-push-handler lima-push-manager: FAKE_FOR=COMPLETION
```
Comment (luphaz) [inline Makefile:53]: This aims to provide bashcompletion or similar as they are parsing the Makefile, not necessarily the templates

File: Makefile (hunk 1 above)
Comment (ptnapoleon) [reply Makefile:65]: This would be helpful as a direct comment right above here

**Review by ptnapoleon — APPROVED** [review]
works for me? The Makefile was getting fairly verbose

File: Makefile (hunk 1 above)
Comment (luphaz) [reply Makefile:65]:
```suggestion
# $(1) is the target name: injector|handler|manager
$(1): BINARY_NAME=$(1)
```

**Pull request merged**